src/query/service/src/interpreters/interpreter_user_stage_remove.rs

```diff
@@ -68,19 +70,17 @@ impl Interpreter for RemoveUserStageInterpreter {
             files: None,
             pattern,
         };
-        let files: Vec<String> = files_info
-            .list(&op, thread_num, None)
-            .await?
-            .into_iter()
-            .map(|file_with_meta| file_with_meta.path)
-            .collect::<Vec<_>>();
+        let files = files_info.list_stream(&op, thread_num, None).await?;
 
         let table_ctx: Arc<dyn TableContext> = self.ctx.clone();
         let file_op = Files::create(table_ctx, op);
 
+        let mut chunks = files.chunks(REMOVE_BATCH);
```

Comment:
s3 can remove at most 1k files in one requests. We can reduce the batch size a bit.

Reply:
done